### PR TITLE
Add EIP: Token Converter

### PR DESCRIPTION
**Pull request by Dexaran**

### CODE REVIEW

**Comment by Dexaran** [comment]
I am proposing a cross-standard operability for tokens i.e. a method of transforming ERC-20 tokens to ERC-223 and back to ERC-20

EIP was originally named "Token Standard Converter"

Description "Smart-contract service that converts token of one standard to another"

The Walidator told me "standard" is considered a bad word here and I renamed it to "Smart-contract service that converts token of one ERC version to another" but the original name is more intuitive for token developers, I think

**Review by bitcoinfinancier — APPROVED** [review]

File: EIPS/eip-0.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
+---
+eip: 0
```
[the remark this answers is not in the document]
Comment (Dexaran) [reply EIPS/eip-0.md]: Sure
Done

**Comment by sgitt-vassky** [comment]
It can be better to rename this eip to `Interoperability of erc-20 and erc-223`

because it focuses on this two standards not tokens in general and 'token converter' name does not explain anything tbh

File: EIPS/eip-7417.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+There are multiple token standards on Ethereum chain currently. This EIP introduces a concept of cross-standard interoperability by creating a service that allows [ERC-20](./eip-20.md) tokens to be upgraded to [ERC-223](./eip-223.md) tokens anytime. [ERC-223](./eip-223.md) tokens can be converted back to [ERC-20](./eip-20.md) version without any restrictions to avoid any problems with backwards compatibility and allow different standards to co-exist and become interoperable and interchangeable.
+
+## Motivation
+
+The main motivation behind this EIP is to formalize the implementation of a service that performs token conversion and provide the address of the contract on mainnet in order to avoid potentially unsafe implementation from spreading.
+
+The idea is to record the contract address here after the deployment of the contract for everyone to publicly access and use.
+
+## Specification
+
```
Comment (SamWilsn) [inline EIPS/eip-7417.md:25]: EIPs, generally, are for standards that require multiple parties to build compatible implementations.

From my (admittedly quick) read, it sounds like there will be exactly one Token Converter and any ERC-20 contract can upgrade using it.

So I have to ask, who are the multiple implementers that need to interoperate? Couldn't you just deploy this service, and anyone could use it? If so, I don't think this proposal needs an EIP.

**Review by SamWilsn — CHANGES_REQUESTED** [review]

File: EIPS/eip-7417.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-07-27
+requires: 20, 165, 223
+---
+
+## Abstract
+
+There are multiple token standards on Ethereum chain currently. This EIP introduces a concept of cross-standard interoperability by creating a service that allows [ERC-20](./eip-20.md) tokens to be upgraded to [ERC-223](./eip-223.md) tokens anytime. [ERC-223](./eip-223.md) tokens can be converted back to [ERC-20](./eip-20.md) version without any restrictions to avoid any problems with backwards compatibility and allow different standards to co-exist and become interoperable and interchangeable.
```
[the remark this answers is not in the document]
Comment (Dexaran) [reply EIPS/eip-7417.md:16]: Added the description https://github.com/Dexaran/EIPs/commit/ebef0f9898c3c827529f6996a1503ac8d5360861

File: EIPS/eip-7417.md (hunk 2 above)
Comment (Pandapip1) [reply EIPS/eip-7417.md:25]: > From my (admittedly quick) read, it sounds like there will be exactly one Token Converter and any ERC-20 contract can upgrade using it.

There is precedent for this being an ERC-category EIP. I forget the EIP number, but there was one that used its EIP number repeating as the r, v, and s values to create an instance of a contract at a predetermined address across all chains.

**Comment by Dexaran** [comment]
@SamWilsn @Pandapip1 can we merge it? Or what is expected from my side?

**Review by Pandapip1 — APPROVED** [review]
The styling definitely deviates a lot from most new EIPs, but that can be fixed and doesn't block this EIP. I'll submit a PR once this is merged. If I forget, @Dexaran please ping me.

**Comment by Dexaran** [comment]
@Pandapip1 the PR was not auto-merged for some reason.

**Pull request merged**